Switch to better implementation of priming images for goldens

coreysprague opened the pull request:
Fixes #36

coreysprague commented on packages/golden_toolkit/example/test/example_test.dart:
```diff
@@ -85,6 +85,24 @@ void main() {
       );
     });
+
+    group('GoldenBuilder examples of accessibility testing', () {
```
re-instated... I accidentally nuked this a couple days ago...

mZadorskii commented on packages/golden_toolkit/lib/src/testing_tools.dart:
```diff
@@ -249,7 +252,52 @@ Future<void> compareWithGolden(
+  });
+}
+
+/// A function that waits for all [Image] widgets found in the widget tree to finish decoding.
```
May be some callout that currently supported:
**Images, Decorations**

mZadorskii commented on packages/golden_toolkit/lib/src/widget_tester_extensions.dart:
```diff
@@ -3,10 +3,22 @@ import 'dart:ui';
+  /// in Goldens. The implementation of this can be configured as part of GoldenToolkitConfiguration
+  ///
+  /// If you have assets that are not loading with this implementation, please file an issue and we will explore solutions.
+  Future<void> waitForAssets() {
```
Future<void> waitForAssets() =>  Future<void> waitForAssets() 
:)

coreysprague commented:
@christian-muertz I merged your PR into here and added the following:
- primed BoxDecoration images
- added tests for the specific priming funcs
- added an extension on WidgetTester to expose the waitForAssets() method
- made your new method the default.

Thanks again for your help!

christian-muertz commented:
> @christian-muertz I merged your PR into here and added the following:
> 
> - primed BoxDecoration images
> 
> - added tests for the specific priming funcs
> 
> - added an extension on WidgetTester to expose the waitForAssets() method
> 
> - made your new method the default.
> 
> 
> 
> Thanks again for your help!
> 
> 
 
I left a comment at the new impl.

coreysprague commented:
@christian-muertz I don't see the comment. Can you re-post?

christian-muertz commented:
Just forgot to finish the review

coreysprague commented:
@christian-muertz you are correct. I'll throw a quick PR up to resolve. I also updated our repo to use this and found 2 out of about 500 goldens that had issues, so I'm going to dig in to try to figure out the gaps.
